src/core/shape_inference/include/strided_slice_shape_inference.hpp

```diff
@@ -65,7 +65,7 @@ std::vector<TRShape> shape_infer(const StridedSlice* op,
     std::optional<std::vector<int64_t>> strides;
     if (input_shapes.size() > 3) {
         strides = get_input_const_data_as<TRShape, int64_t>(op, 3, ta);
-    } else if (begin) {
```

Comment:
Why this check has been removed? 
`begin->` is used just below, so it should be ensured:
https://github.com/openvinotoolkit/openvino/blob/66f79c0b39a45afb006725b828b0355bd65a7b50/src/core/shape_inference/include/strided_slice_shape_inference.hpp#L70